### PR TITLE
Devicemapper support!

### PR DESCRIPTION
**Pull request by iancmcc**
Currently working through Godeps issues, but the code is good.

### CODE REVIEW

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

File: commons/docker/api.go [hunk 1]
```diff
@@ -724,12 +724,15 @@ func pullImage(repo, registry, tag string) error {
 
 // PushImage pushes an image by repotag to local registry, e.g., zenoss/devimg, from the local docker repository
 func PushImage(repotag string) error {
-	iid, err := commons.ParseImageID(repotag)
-	if err != nil {
-		return err
+	return nil
```
Comment (smousa) [inline commons/docker/api.go]: Why did you comment this?

File: dfs/snapshot.go [hunk 2]
```diff
@@ -88,21 +88,26 @@ func (dfs *DistributedFilesystem) Snapshot(tenantID string, description string)
 		return "", err
 	}
 
+	tagID := time.Now().UTC().Format(timeFormat)
+	label := fmt.Sprintf("%s_%s", tenantID, tagID)
+
+	mdPath := snapshotVolume.SnapshotMetadataPath(label)
+	if err := os.MkdirAll(mdPath, 0755); err != nil {
```
Comment (smousa) [inline dfs/snapshot.go]: I don't think this will work if you have btrfs filesystem

File: volume/btrfs/btrfs.go [hunk 3]
```diff
@@ -34,6 +35,13 @@ const (
 	DriverName = "btrfs"
 )
 
+var (
+	ErrBtrfsCommand      = errors.New("error running btrfs command")
+	ErrCreatingSubvolume = errors.New("could not create subvolume")
+	ErrInvalidLabel      = errors.New("invalid label")
+	ErrListingSnapshots  = errors.New("couldn't list snapshots")
+)
```
Comment (smousa) [inline volume/btrfs/btrfs.go]: suggestion: perhaps you should prepend your errors with "btrfs:"

File: volume/btrfs/btrfs.go [hunk 4]
```diff
@@ -34,6 +35,13 @@ const (
 	DriverName = "btrfs"
 )
 
+var (
+	ErrBtrfsCommand      = errors.New("error running btrfs command")
```
Comment (smousa) [inline volume/btrfs/btrfs.go]: this should perhaps be a type where you can include the command being called along with the output of the command that implement error

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

**Pull request merged**